application/src/main/scala/com/azavea/franklin/api/Server.scala

```diff
@@ -34,15 +33,15 @@ import java.util.concurrent.Executors
 
 object Server extends IOApp {
 
-  val franklinIO: ContextShift[IO] = IO.contextShift(
-    ExecutionContext.fromExecutor(
+  val franklinContextShift: Resource[IO, ContextShift[IO]] = Resource.make(
+    IO.delay {
       Executors.newCachedThreadPool(
         new ThreadFactoryBuilder().setNameFormat("raster-io-%d").build()
       )
-    )
-  )
-
-  override implicit val contextShift: ContextShift[IO] = franklinIO
+    }
+  )(pool => IO.delay(pool.shutdown)) map { service =>
+    IO.contextShift(ExecutionContext.fromExecutor(service))
+  }
```

Comment:
It makes sense, :+1: that's what was happening. 

Can we try [IOApp.WithContext](https://github.com/typelevel/cats-effect/blob/90f600fa16c241bbd072b4e0cd7d28786abfeee0/benchmarks/shared/src/main/scala/cats/effect/benchmarks/ECBenchmark.scala#L41-L43) here? It will require the execution context to be a resource and this will do the same trick but in a bit more clean fashion.

see the discussion for the context: https://github.com/typelevel/cats-effect/issues/337

Reply:
2075fa4

I didn't know about this one, but it worked great 😎